### PR TITLE
Terms Signature endpoints for kamino-sdk and hubble-sdk

### PR DESCRIPTION
**Pull request by mihalex98**
Added the following:

**kamino-sdk** 

- updated code-gen only with changes for termsSignature
- `getUserTermsSignatureIx` - to get the ix that creates the terms signature for a given user 
- `getUserTermsSignatureState` - to get the signatureState for a given user
- integration test case -> creates a user and sgins using ed25519 message signature (with user secretKey and a basic message) -third party library used to replicate signature as `signMessage` from solana-wallet-adapter didn't work locally  -> sends the transaction to store signature on-chain -> retrieves sigantureState and checks that on-chain siganture matches

**hubble**

- added helper code to interact with sign_terms instruction of the borrowing program
- `getUserTermsSignatureIx` - to get the ix that creates the terms signature for a given user 
- `getUserTermsSignatureState` - to get the signatureState for a given user
- modified constructor to accept (as optional arg) the programId so that we can test hubble-sdk locally (using a localnet program and programId)
- new integration test file and new utils test file that performs the same test as in kamino on localnet

### CODE REVIEW

File: packages/hubble-sdk/src/Hubble.ts [hunk 1]
```diff
@@ -658,6 +663,50 @@ export class Hubble {
...
+  */
+  async getUserTermsSignatureState(owner: PublicKey): Promise<TermsSignature | null> {
+    const pdaSeed = [
+      Buffer.from("signature"),
+      owner.toBuffer(),
+    ];
+    const [signatureStateKey, _signatureStateBump] =
+      PublicKey.findProgramAddressSync(pdaSeed, this._config.borrowing.programId);
+
+    const signatureState = await TermsSignature.fetch(this._connection, signatureStateKey, this._config.borrowing.programId);
```
Comment (peroxy) [inline packages/hubble-sdk/src/Hubble.ts:706]: Can be simplified with:

```ts
return TermsSignature.fetch(this._connection, signatureStateKey, this._config.borrowing.programId);
```

File: packages/hubble-sdk/tests/integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const message = Uint8Array.from([0xab, 0xbc, 0xcd, 0xde]);
+    const signature = ed25519.sign(message, owner.secretKey);
+  
+    // initialize signature
+    const signTermsIx = await hubble.getUserTermsSignatureIx(owner.publicKey, signature);
+    const tx = new Transaction();
+    tx.add(signTermsIx);
+    const sig = await sendTransactionWithLogs(connection, tx, owner.publicKey, [owner]);
+  
+    // assert there is one strat that is STABLE
```
Comment (peroxy) [inline packages/hubble-sdk/tests/integration.test.ts:38]: I think you forgot to actually assert anything in this test :)

File: packages/kamino-sdk/src/Kamino.ts [hunk 3]
```diff
@@ -1635,6 +1636,50 @@ export class Kamino {
...
+   */
+  async getUserTermsSignatureState(owner: PublicKey): Promise<TermsSignature | null> {
+    const pdaSeed = [
+      Buffer.from("signature"),
+      owner.toBuffer(),
+    ];
+    const [signatureStateKey, _signatureStateBump] =
+      PublicKey.findProgramAddressSync(pdaSeed, this._kaminoProgramId);
+
+    const signatureState = await TermsSignature.fetch(this._connection, signatureStateKey);
```
Comment (peroxy) [inline packages/kamino-sdk/src/Kamino.ts:1651]: Can be simplified as well

**Review by peroxy — APPROVED** [review]
Looks good, just a few small comments!

**Review by y2kappa — APPROVED** [review]
LGTM

**Comment by y2kappa** [comment]
need to fix the CI

**Comment by mihalex98** [comment]
> need to fix the CI

code is building now. test dockerfile will keep failing until this is deployed so the dumped .so in the container (used to test) is up to date

**Review by y2kappa — APPROVED** [review]

**Pull request merged**